java/common/exception/ErrorMessage.java

```diff
@@ -62,17 +62,19 @@
     public static final ErrorMessage INVALID_RULE_WHEN_CONTAINS_DISJUNCTION =
             new ErrorMessage(21, "Rule '%s' 'when' contains a disjunction.");
     public static final ErrorMessage INVALID_RULE_THEN =
-            new ErrorMessage(22, "Rule '%s' 'then' must strictly be either an attribute ownership, or a relation: '%s'.");
+            new ErrorMessage(22, "Rule '%s' 'then' must be exactly one attribute ownership, or exactly one relation: '%s'.");
+    public static final ErrorMessage INVALID_RULE_THEN_HAS =
+            new ErrorMessage(23, "Rule '%s' 'then' is trying to assign both an attribute type and a variable attribute value: '%s'.");
```

Comment:
Putting the `then` clause that the user wrote at the right end of the message implies that that is what you're suggesting to be the right way, as opposed to pointing out that that is wrong. A better way is to have the content of the `then` clause to be mentioned right the start, such as: `"Rule '%s' 'then' ('%s') ..."`.

Reply:
Fixed those error messages.